assignment4-Due 11.12.Rmd

```diff
@@ -0,0 +1,230 @@
+CORF1 <- cor(DF1)
+
+corrplot(CORF1, order="AOE", method="circle", tl.pos="lt", type="upper",      
+tl.col="black", tl.cex=0.6, tl.srt=45, 
+        addCoef.col="black", addCoefasPercent = TRUE,
+        sig.level=0.50, insig = "blank")
+
+# Scale and center the data 
+DF2 <- scale(DF1, center = TRUE)
+pcaF <- prcomp(DF2, scale = TRUE)
+
+# Look up for which components to keep and which to drop
+pcaF$sdev
+
+pcaF$sdev^2
+
+summary(pcaF)
+
+plot(pcaF, type = "lines")
+
+# Keep 1:5
+DF3 <- pcaF$x[, c(1:5)]
+
+# Generate pairwise correlations for refined data
+CORF2 <- cor(DF3)
+
+corrplot(CORF2, order="AOE", method="circle", tl.pos="lt", type="upper",      
+tl.col="black", tl.cex=0.6, tl.srt=45, 
+        addCoef.col="black", addCoefasPercent = TRUE,
+        sig.level=0.50, insig = "blank")
+
+# Calculate values
+pcaF$rotation
+DRF1 <- t(pcaF$rotation)
+DRF2 <- pcaF$x %*% DRF1
+corrgram(DRF2)
+
+# Using biplot to get a visualization of which programs "go together". (A little bit hard to see because of too many variables.)
+
+biplot(pcaF)
+
+# Another approach: heatmap
+heatmap(DRF2)
+
+```
+```{r}
+### Determine which program goes together: Using descending order to rank programs by PCs
+
+ # Make all eigenvectors positive
+loadings_rank <- abs(pcaF$rotation)
+
+# Computes each row as a proportion of the column.
+loadings_rank_df <- as.data.frame(sweep(loadings_rank, 2, colSums(loadings_rank), "/")) 
+
+# Generating a components*program matrix
+components_by_program <- data.frame(initialise = matrix(NA, nrow = 68, ncol = 1))
+
+# Get all PCs names
+PCs <- colnames(loadings_rank_df)
+PCs
+
+# Get all programs names
+programN <- rownames(loadings_rank_df)
+
+# Get the components*program matrix
+loadings_rank_df <- cbind(programN, loadings_rank_df)
+
+# Using for loop to generate descending order of program in terms of PCs 
+library(tidyr)
+library(dplyr)
+for (column in PCs) {
+  sorted_programs <- arrange(loadings_rank_df, desc(!!sym(column)))
+  components_by_program <- cbind(components_by_program, sorted_programs, stringsAsFactors = FALSE)
+}
+
+# Results
+components_by_program <- components_by_program[, -1]
+components_by_program 
+
+
+### Comments: 
+# Group 1( Leadership and Orgnizational psychology related programs): Change.Leadership, Economics.and.Education, Education.Policy, School.Principals, Social.Organizational.Psychology, Politics
+# Group 2 (Special Education related programs):Diabetes.Education, Music, Instructional.Technology.and.Media, Nursing, Deaf.and.Hard.of.Hearing,  Cognitive.Science
+# Group 3 (Neoro, Physical and Health related programs): Kinesiology, 	Clinical.Psychology, Physiology, Neuroscience, 	Health.Education, Nursing 
+# Group 4 (Learning analytics, Mathematics, Educational Technology related programs): Learning.Analytics, Cognitive.Science, Mathematics, Design.and.Development.of.Digital.Games, Measurement..Evaluation.and.Statistics, Instructional.Technology.and.Media
+```
```

Comment:
Very nice work! I appreciate your effort in analyzing and interpreting the meaning of principle components. A minor suggestion: try to only view the head of a dataset in your code, otherwise the published html page would be very redundant and obscure your main analysis.